specification/deviceregistry/resource-manager/Microsoft.DeviceRegistry/preview/2023-11-01-preview/asset.json

```diff
@@ -0,0 +1,663 @@
+                "default": "none",
+                "description": "An indication of how the data point should be mapped to OpenTelemetry."
+              },
+              "dataPointConfiguration": {
+                "type": "string",
+                "description": "Protocol-specific configuration for the data point. For OPC UA, this could include configuration like, publishingInterval, samplingInterval, and queueSize."
+              }
+            }
+          },
+          "x-ms-identifiers": [],
+          "description": "Array of data points that are part of the asset. Each data point can reference an asset type capability and have per-data point configuration. See below for more details for the definition of the dataPoints element."
+        },
+        "events": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "required": [
+              "eventNotifier"
+            ],
+            "properties": {
+              "name": {
+                "type": "string",
+                "description": "The name of the event."
+              },
+              "eventNotifier": {
+                "type": "string",
+                "description": "The address of the notifier of the event in the asset (e.g. URL) so that a client can access the event on the asset."
+              },
+              "capabilityId": {
+                "type": "string",
+                "description": "The path to the type definition of the capability (e.g. DTMI, OPC UA information model node id, etc.), for example dtmi:com:example:Robot:_contents:__prop1;1."
+              },
+              "observabilityMode": {
+                "type": "string",
+                "enum": [
+                  "none",
+                  "log"
+                ],
+                "x-ms-enum": {
+                  "name": "EventsObservabilityMode",
+                  "modelAsString": true,
+                  "values": [
+                    {
+                      "name": "None",
+                      "value": "none"
+                    },
+                    {
+                      "name": "Log",
+                      "value": "log"
+                    }
+                  ]
+                },
+                "default": "none",
+                "description": "An indication of how the event should be mapped to OpenTelemetry."
+              },
+              "eventConfiguration": {
+                "type": "string",
+                "description": "Protocol-specific configuration for the event. For OPC UA, this could include configuration like, publishingInterval, samplingInterval, and queueSize."
+              }
+            }
+          },
+          "x-ms-identifiers": [],
+          "description": "Array of events that are part of the asset. Each event can reference an asset type capability and have per-event configuration. See below for more details about the definition of the events element."
+        },
+        "status": {
```

Comment:
Is this `status` object with `errors` a common ARM practice? I'm mostly curious as to why we can't leverage the [ARM error response content](https://github.com/Azure/typespec-azure/tree/main/packages/typespec-providerhub-templates) ?

Reply:
This is based on the status sync protocl for that has been reviewed with the Akri and OPCUA teams. The status sync protocol is utilized by brokers on the edge to indicate asset status back to the cloud. This is a mechanism for components on the edge to report back state on an asset asynchronously via ARC.